teos/gatekeeper.py

```diff
@@ -83,20 +89,46 @@ class Gatekeeper:
             database.
         registered_users (:obj:`dict`): A map of ``user_pk:user_info``.
         lock (:obj:`Lock`): A lock object to lock access to the Gatekeeper on updates.
-
     """
 
     def __init__(self, user_db, block_processor, subscription_slots, subscription_duration, expiry_delta):
         self.subscription_slots = subscription_slots
         self.subscription_duration = subscription_duration
         self.expiry_delta = expiry_delta
+        self.block_queue = Queue()
+        self.expired_users_cache = []
         self.block_processor = block_processor
         self.user_db = user_db
         self.registered_users = {
             user_id: UserInfo.from_dict(user_data) for user_id, user_data in user_db.load_all_users().items()
         }
         self.lock = Lock()
 
+        # Starts a child thread to take care of expiring subscriptions
+        Thread(target=self.manage_subscription_expiry, daemon=True).start()
+
+    def manage_subscription_expiry(self):
+        """
+        Manages the subscription expiry of the registered users. When a subscription reaches it end, it moves it to
```

Comment:
Typo, "its end"

Reply:
Changes the description here, so it does not apply anymore.